origin-dapp/src/components/conversation.js

```diff
@@ -233,78 +275,81 @@ class Conversation extends Component {
     }
   }
 
+  formatOfferMessage(info) {
+    const { listing = {} } = this.state
+    const { users, intl, smallScreenOrDevice, withListingSummary } = this.props
+
+    if (smallScreenOrDevice || !withListingSummary) return
+
+    const { returnValues = {}, event, timestamp } = info
+    const partyAddress = formattedAddress(returnValues.party)
+    const user = users.find((user) => formattedAddress(user.address) === partyAddress)
+    const userName = abbreviateName(user)
+    const party = userName || truncateAddress(returnValues.party)
+
+    const offerMessages = {
+      'OfferCreated': `${party} ${intl.formatMessage(
+        this.intlMessages.offerCreated
+      )}`,
+      'OfferWithdrawn': `${party} ${intl.formatMessage(
```

Comment:
@Natasha08 this needs to handle the rejection/withdrawal distinction. They are the same at the contract level, so we have to interpret the perspective. See [here](https://github.com/OriginProtocol/origin/commit/9db3ce08a1d231ae922882e66fd36adb88321743#diff-77a57a9603c69e6c5d5af8d7773c0d7eR58).

Reply:
@micahalcorn Okay I'll take a look and update the code